aws/logs_monitoring/logs.py

```diff
@@ -96,36 +99,33 @@ def compileRegex(rule, pattern):
             )
 
 
-include_regex = compileRegex("INCLUDE_AT_MATCH", INCLUDE_AT_MATCH)
-exclude_regex = compileRegex("EXCLUDE_AT_MATCH", EXCLUDE_AT_MATCH)
-
-
-# should only be called when INCLUDE_AT_MATCH and/or EXCLUDE_AT_MATCH exist
-def filter_logs(logs):
+def filter_logs(logs, include_pattern=None, exclude_pattern=None):
     """
     Applies log filtering rules.
     If no filtering rules exist, return all the logs.
     """
-    if INCLUDE_AT_MATCH is None and EXCLUDE_AT_MATCH is None:
+    if include_pattern is None and exclude_pattern is None:
         return logs
     # Add logs that should be sent to logs_to_send
     logs_to_send = []
     for log in logs:
-        if EXCLUDE_AT_MATCH is not None or INCLUDE_AT_MATCH is not None:
+        if exclude_pattern is not None or include_pattern is not None:
             logger.debug("Filtering log event:")
             logger.debug(log)
         try:
-            if EXCLUDE_AT_MATCH is not None:
+            if exclude_pattern is not None:
                 # if an exclude match is found, do not add log to logs_to_send
-                logger.debug(f"Applying EXCLUDE_AT_MATCH: {EXCLUDE_AT_MATCH}")
+                logger.debug(f"Applying exclude pattern: {exclude_pattern}")
+                exclude_regex = compileRegex("EXCLUDE_AT_MATCH", exclude_pattern)
```

Comment:
I moved the regex compilation into this function to make it more testable, rather than relying on global variables for the compiled regex. This should not negatively impact performance because even though the compile function is called for every log line, the `re` module automatically caches the last 512 compiled regular expressions.

Per the `re` docs: "The compiled versions of the most recent patterns passed to re.compile() and the module-level matching functions are cached, so programs that use only a few regular expressions at a time needn’t worry about compiling regular expressions."

([Here](https://github.com/python/cpython/blob/master/Lib/re.py#L284) is the code for the cache)